marklogic-data-hub-spark-connector/src/main/java/com/marklogic/hub/spark/sql/sources/v2/writer/HubDataSourceWriter.java

```diff
@@ -101,6 +100,48 @@ private boolean atLeastOneWriteFailed(WriterCommitMessage[] messages) {
         return false;
     }
 
+    /**
+     * Validate the URI template if present in the options.
+     *
+     * A valid URI template:
+     * - Has no missing opening or closing curly brackets.
+     * - Each token has >0 characters.
+     * - Has no nested curcly brackets.
+     */
+    private void validateUriTemplateIfPresent() {
+        String uriTemplate = options.get("uritemplate");
+        if ( uriTemplate == null ) {
+            return;
+        }
+        boolean inToken = false;
+        int tokenSize = 0;
+        char[] chars = uriTemplate.toCharArray();
+        for ( char ch : chars ) {
+            if ( ch == '}' ) {
+                if ( !inToken ) {
+                    throw new RuntimeException("Closing curly bracket found without opening bracket. uritemplate=[" + uriTemplate + "].");
```

Comment:
For consistency with my proposal for error messages in the DS endpoint, let's do a format of:

```
throw new IllegalArgumentException("Cannot apply uriTemplate: " + uriTemplate + "; closing curly bracket found without opening bracket");
```

IllegalArgumentException is a bit more descriptive than RuntimeException. And we don't have a precedent for using "[" and "]" to surround user-provided inputs (it's a nice pattern, we just don't do it elsewhere).

Reply:
ready for commit.